src/metric.jl

```diff
@@ -99,25 +99,36 @@ function Base.show(io::IO, dem::DenseEuclideanMetric)
 end
 
 """
-    RankUpdateEuclideanMetric{T,M} <: AbstractMetric
+    RankUpdateEuclideanMetric{T,AM,AB,AD,F} <: AbstractMetric
 
 A Gaussian Euclidean metric whose inverse is constructed by rank-updates.
 
+# Fields
+
+$(TYPEDFIELDS)
+
 # Constructors
 
     RankUpdateEuclideanMetric(n::Int)
+    RankUpdateEuclideanMetric(M⁻¹, B, D)
 
-Construct a Gaussian Euclidean metric of size `(n, n)` with inverse of `M⁻¹`.
+ - Construct a Gaussian Euclidean metric of size `(n, n)` with `M⁻¹` being diagonal matrix.
+ - Construct a Gaussian Euclidean metric of `M⁻¹`, where `M⁻¹` should be a full rank positive definite matrix,
+    and `B` `D` must be chose so that the Woodbury matrix `W = M⁻¹ + B D B^\\mathrm{T}` is positive definite.
 
 # Example
 
 ```julia
 julia> RankUpdateEuclideanMetric(3)
 RankUpdateEuclideanMetric(diag=[1.0, 1.0, 1.0])
 ```
+
+# References
+
+ - Ben Bales, Arya Pourzanjani, Aki Vehtari, Linda Petzold, Selecting the Metric in Hamiltonian Monte Carlo, 2019
 """
 struct RankUpdateEuclideanMetric{T,AM<:AbstractVecOrMat{T},AB,AD,F} <: AbstractMetric
```

Comment:
More a question than a request: Is there a reason why `AM` can be a Vector? Also, is it intentional that `AB` and `AD` don't have to have the same element type?